File: updated_code/code/java/160-intersection-of-two-linked-lists.java
/**
 * This class provides a solution to find the intersection of two linked lists.
 */
public class Solution {
    /**
     * Returns the node at which the two linked lists intersect.
     * If the two linked lists do not intersect, returns null.
     *
     * @param headA the head of the first linked list
     * @param headB the head of the second linked list
     * @return the node at which the two linked lists intersect
     */
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        if (headA == null || headB == null) {
            return null;
        }

        ListNode pItemA = headA;
        ListNode pItemB = headB;

        while (pItemA != pItemB) {
            pItemA = pItemA != null ? pItemA.next : headB;
            pItemB = pItemB != null ? pItemB.next : headA;
        }

        return pItemA;
    }
}

File: test_data/code/python/160-intersection-of-two-linked-lists.py
class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        if headA is None or headB is None:
            return None

        pA = headA
        pB = headB

        while pA is not pB:
            if pA is not None:
                pA = pA.next
            else:
                pA = headB

            if pB is not None:
                pB = pB.next
            else:
                pB = headA

        return pA